.github/workflows/pr_shelter.yml

```diff
@@ -1,55 +1,148 @@
+        fi;
+        echo "inclavare_test=$inclavare_test" >> $GITHUB_ENV
+
+    - name: Install ubuntu dependency
+      if: ${{ contains(matrix.tag, 'ubuntu') }}
+      run: |
+        docker exec $inclavare_test bash -c 'apt-get update -y && apt-get install -y wget gnupg;
+        echo "deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic main" | tee /etc/apt/sources.list.d/intel-sgx.list && wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add -;
+        apt-get update -y && apt-get install -y libsgx-dcap-quote-verify libsgx-dcap-default-qpl libsgx-dcap-ql-dev libsgx-uae-service;
+        cp -rf /root/inclavare-containers/${{ matrix.tag }}/enclave-tls /opt'
+
+    - name: Install centos dependency
+      if: ${{ contains(matrix.tag, 'centos') }}
       run: |
-        docker exec $rune_test bash -c "cd /root/inclavare-containers-test/shelter;
-        make -j1;
-        ./shelter remoteattestation"
+        docker exec $inclavare_test bash -c 'cd /root/inclavare-containers/${{ matrix.tag }};
+        yum -y install yum-utils wget iptables;
+        wget -c https://download.01.org/intel-sgx/sgx-linux/2.13/distro/centos8.2-server/sgx_rpm_local_repo.tgz;
+        tar xzf sgx_rpm_local_repo.tgz;
+        yum-config-manager --add-repo sgx_rpm_local_repo;
+        yum makecache;
+        yum install --nogpgcheck -y libsgx-dcap-quote-verify libsgx-dcap-default-qpl libsgx-dcap-ql-devel libsgx-uae-service;
+        rm -f sgx_rpm_local_repo.tgz;
+        mv /root/inclavare-containers/${{ matrix.tag }}/enclave-tls /opt'
+
+    - name: Install alinux dependency
+      if: ${{ contains(matrix.tag, 'alinux') }}
+      run: docker exec $inclavare_test bash -c 'cd /root/inclavare-containers/${{ matrix.tag }};
+        yum -y install yum-utils wget tar procps alinux-release-experimentals  protobuf-c-devel;
+        wget -c https://mirrors.openanolis.org/inclavare-containers/alinux2/sgx_rpm_local_repo.tar.gz;
+        tar xzf sgx_rpm_local_repo.tar.gz;
+        yum-config-manager --add-repo file:/root/inclavare-containers/${{ matrix.tag }}/sgx_rpm_local_repo;
+        yum makecache;
+        yum install --nogpgcheck -y libsgx-dcap-quote-verify libsgx-dcap-default-qpl libsgx-dcap-ql-devel libsgx-uae-service;
+        rm -f sgx_rpm_local_repo.tar.gz;
+        mv /root/inclavare-containers/${{ matrix.tag }}/enclave-tls /opt'
+
+    - id: random-port-generator
+      uses: ./.github/actions/generate-random-port
+
+    - run: echo random-port ${{ steps.random-port-generator.outputs.random-port }}
+      shell: bash
+
+    - name: Run enclave tls server for null quote
+      if: ${{ contains(matrix.sgx, 'SGX1') }}
+      run: docker exec $inclavare_test bash -c 'cd /opt/enclave-tls/bin/ && ./enclave-tls-server -a nullquote' &
```

Comment:
`null quote` can run in SGX1 too. Please delete line 112 and line 116.

Reply:
Do you mean SGX2?

Reply:
This condition just specify run null quote in SGX1 machine.

Reply:
Support to test null quote instance for SGX1 & SGX2